include/nbl/builtin/hlsl/workgroup2/arithmetic_config.hlsl

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_WORKGROUP2_ARITHMETIC_CONFIG_INCLUDED_
+#define _NBL_BUILTIN_HLSL_WORKGROUP2_ARITHMETIC_CONFIG_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace workgroup2
+{
+
+namespace impl
+{
+template<uint16_t WorkgroupSizeLog2, uint16_t SubgroupSizeLog2>
+struct virtual_wg_size_log2
+{
+    static_assert(WorkgroupSizeLog2>=SubgroupSizeLog2, "WorkgroupSize cannot be smaller than SubgroupSize");
+    // static_assert(WorkgroupSizeLog2<=SubgroupSizeLog2+4, "WorkgroupSize cannot be larger than SubgroupSize*16");
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t levels = conditional_value<(WorkgroupSizeLog2>SubgroupSizeLog2),uint16_t,conditional_value<(WorkgroupSizeLog2>SubgroupSizeLog2*2+2),uint16_t,3,2>::value,1>::value;
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t value = mpl::max_v<uint32_t, WorkgroupSizeLog2-SubgroupSizeLog2, SubgroupSizeLog2>+SubgroupSizeLog2;
+    // must have at least enough level 0 outputs to feed a single subgroup
+};
+
+template<class VirtualWorkgroup, uint16_t BaseItemsPerInvocation, uint16_t WorkgroupSizeLog2, uint16_t SubgroupSizeLog2>
+struct items_per_invocation
+{
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t ItemsPerInvocationProductLog2 = mpl::max_v<int16_t,WorkgroupSizeLog2-SubgroupSizeLog2*VirtualWorkgroup::levels,0>;
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t value0 = BaseItemsPerInvocation;
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t value1 = uint16_t(0x1u) << conditional_value<VirtualWorkgroup::levels==3, uint16_t,mpl::min_v<uint16_t,ItemsPerInvocationProductLog2,2>, ItemsPerInvocationProductLog2>::value;
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t value2 = uint16_t(0x1u) << mpl::max_v<int16_t,ItemsPerInvocationProductLog2-2,0>;
+};
+
+// explicit specializations for cases that don't fit
+#define SPECIALIZE_VIRTUAL_WG_SIZE_CASE(WGLOG2, SGLOG2, LEVELS, VALUE) template<>\
+struct virtual_wg_size_log2<WGLOG2, SGLOG2>\
+{\
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t levels = LEVELS;\
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t value = VALUE;\
+};\
+
+SPECIALIZE_VIRTUAL_WG_SIZE_CASE(11,4,3,12);
+SPECIALIZE_VIRTUAL_WG_SIZE_CASE(7,7,1,7);
+SPECIALIZE_VIRTUAL_WG_SIZE_CASE(6,6,1,6);
+SPECIALIZE_VIRTUAL_WG_SIZE_CASE(5,5,1,5);
+SPECIALIZE_VIRTUAL_WG_SIZE_CASE(4,4,1,4);
+SPECIALIZE_VIRTUAL_WG_SIZE_CASE(3,3,1,3);
+SPECIALIZE_VIRTUAL_WG_SIZE_CASE(2,2,1,2);
```

Comment:
the reason they don't work out the box is because your `value` is defined in terms of `max_v<uint` and not `max_v<int`

the `levels` was always correct